tcms/testcases/models.py

```diff
@@ -522,29 +520,27 @@ def mail(self, template, subject, context={}, to=[], request=None):
         mailto(template, subject, to, context, request)
 
     def remove_bug(self, bug_id, run_id=None):
-        sql = SQL.REMOVE_BUG
-        args = [bug_id, self.pk]
+        query = TestCaseBug.objects.filter(
+            bug_id=bug_id,
+            case=self.pk
+        )
         if run_id:
-            sql = SQL.REMOVE_BUG_WITH_RUN_ID
-            args.append(run_id)
-        with transaction.atomic():
-            cursor = connection.writer_cursor
-            cursor.execute(sql, args)
+            query = query.filter(case_run=run_id)
+        else:
+            query = query.filter(case_run__isnull=True)
+
+        query.delete()
 
     def remove_component(self, component):
-        with transaction.atomic():
-            cursor = connection.writer_cursor
-            cursor.execute(SQL.REMOVE_COMPONENT, (self.case_id, component.id))
+        # note: cannot use self.component.remove(component) on a ManyToManyField
+        # which specifies an intermediary model so we use the model manager!
+        self.component.through.objects.filter(case=self.pk, component=component.pk).delete()
```

Comment:
Use intermediary table can make it much easy to understand. 

```
TestCaseComponent.objects.filter(case=self, component=component).delete()
```

It's straightforward and no need of comment here.

Same comment to below changes.